contrib/mod_tls.c

```diff
@@ -13005,6 +13004,40 @@ MODRET tls_any(cmd_rec *cmd) {
   return PR_DECLINED(cmd);
 }
 
+MODRET tls_pre_stor(cmd_rec *cmd) {
+#if defined(TLS1_3_VERSION)
+  SSL_SESSION *ssl_session;
+  int ssl_version;
+#endif /* TLS1_3_VERSION */
+
+  if (tls_engine == FALSE) {
+    return PR_DECLINED(cmd);
+  }
+
+  if (!(tls_flags & TLS_SESS_ON_CTRL)) {
+    return PR_DECLINED(cmd);
+  }
+
+  if (ctrl_ssl == NULL) {
+    return PR_DECLINED(cmd);
+  }
+
+# if defined(TLS1_3_VERSION)
+  ssl_session = SSL_get_session(ctrl_ssl);
+  ssl_version = SSL_SESSION_get_protocol_version(ssl_session);
+
+  if (ssl_version != TLS1_3_VERSION) {
+    return PR_DECLINED(cmd);
+  }
+
+  pr_trace_msg(trace_channel, 19,
+    "requesting new TLSv1.3 session ticket before upload");
+  SSL_new_session_ticket(ctrl_ssl);
```

Comment:
This required `SSL_new_session_ticket` API only appeared in OpenSSL 3.x, which means that OpenSSL-1.x sites which encounter this issue won't have a good solution, just the workaround of increasing the ticket/cache lifetime.